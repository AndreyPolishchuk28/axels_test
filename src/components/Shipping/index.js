import React from "react";
import {Arrow, ShippingInfo, Title} from "./ShippingInfo/ShippingInfo";
import {Row, Col, Button, Form} from 'react-bootstrap';
import {Order} from "./Order/Order";
import styled from "styled-components";
import {Link} from "react-router-dom";


export const Shipping = () =>{
    return(
        <Container className='container-fluid'>
            <Row className='d-flex justify-content-center bg'>
                <Col className='bg-white wrapper-shipping-width' md={6} sm={8} xs={10}>
                    <Row>
                        <Col className='d-flex align-items-center shipping-wrapper'>
                            <span className='shipping-color'>Shipping</span>
                            <Arrow/>
                            <span>Billing</span>
                            <Arrow/>
                            <span>Payment</span>
                        </Col>
                    </Row>
                    <Row>
                        <Col className='text-left'>
                            <Title>Shipping Info</Title>
                        </Col>
                    </Row>
                    <Form>
                        <Form.Group>
                            <Form.Label className='recipient mb-3'>Recipient</Form.Label>
                            <Form.Control type="text" placeholder="Full Name" />
                            <Row className='daytime'>
                                <Col md={8} sm={8} xs={8}>
                                    <Form.Control type="text" placeholder="Daytime Phone" />
                                </Col>
                                <Col>
                                    <Form.Text className="text-muted">
                                        For delivery quetions only
                                    </Form.Text>
                                </Col>
                            </Row>
                            <Form.Label className='recipient mb-0'>Address</Form.Label>
                        </Form.Group>
                    </Form>

                    <ShippingInfo/>
                    <Row>
                        <Col md={7} sm={7} xs={7}>
                            <Link to='/billing'>
                                <Button className='button-continue mb-4' variant="primary" type="button" block>Continue</Button>
                            </Link>
                        </Col>
                    </Row>
                </Col>
                <Col className='grey wrapper-order-width' md={5} sm={8} xs={10}>
                    <Order/>
                </Col>
            </Row>
        </Container>
    )
};

export const Container = styled.div`
    background-color: #E6E9F0;
    padding-top: 15px;
    height: 100vh;
`;
