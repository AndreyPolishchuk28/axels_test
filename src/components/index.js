export { Shipping } from './Shipping/ShippingInfo';
export { Billing } from './Shipping/Billing';
export { Payment } from './Shipping/Payment';
export { ThanksPage } from './Shipping/ThanksPage';

