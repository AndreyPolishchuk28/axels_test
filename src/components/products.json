{
  "products": [
    {
      "name": "The Chelsea Boot",
      "color": "black",
      "price": 235,
      "quantity": 1
    },
    {
      "name": "The Twill Snap Backpack",
      "color": "Reverse Denim + Brown leather",
      "price": 65,
      "quantity": 1
    }
  ]
}
