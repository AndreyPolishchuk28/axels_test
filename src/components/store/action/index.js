import {put, take, all} from 'redux-saga/effects'

const GET_PRODUCTS ='GET_PRODUCTS';
export const SET_PRODUCTS = 'SET_PRODUCTS';

export const getProducts = (payload) =>{
    return{
        type: GET_PRODUCTS,
        payload: payload
    }
};

export function* getProductsSaga() {
    while (true){
        try {
            yield take(GET_PRODUCTS);
            const req = yield fetch('https://demo3830727.mockable.io/products');
            const res = yield req.json();
            console.log(res);
            yield put({
                type:SET_PRODUCTS,
                payload: res
            })
        } catch (e) {
            console.log(e);
        }
    }
}


