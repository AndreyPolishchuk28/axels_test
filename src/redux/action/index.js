import {put, take} from 'redux-saga/effects'

const GET_PRODUCTS ='GET_PRODUCTS';
export const SET_PRODUCTS = 'SET_PRODUCTS';
export const SHIPPING_INFO = 'SHIPPING_INFO';

export const getProducts = (payload) =>{
    return{
        type: GET_PRODUCTS,
        payload: payload
    }
};

export const shippingInfo = (payload) =>{
    return{
        type: SHIPPING_INFO,
        payload: payload
    }
};

export function* getProductsSaga() {
    while (true){
        try {
            yield take(GET_PRODUCTS);
            const req = yield fetch('https://demo3830727.mockable.io/products');
            const res = yield req.json();
            yield put({
                type: SET_PRODUCTS,
                payload: res
            })
        } catch (e) {
            console.log(e);
        }
    }
}

export function* shippingInfoSaga() {
    while (true){
        try {
            const {payload} = yield take(SHIPPING_INFO);
            const response = yield fetch('https://demo3830727.mockable.io/info', {
                    method: 'POST',
                    headers: {
                        "Content-Type": "application/json"
                    },
                    body: JSON.stringify(payload)
                }
        );
            console.log(response.status);
        } catch (e) {
            console.log(e);
        }
    }
}


