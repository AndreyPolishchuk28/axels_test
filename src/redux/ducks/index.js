import { put, take, all } from 'redux-saga/effects'

const GET_PRODUCTS ='GET_PRODUCTS';
const SET_PRODUCTS = 'SET_PRODUCTS';
const SHIPPING_INFO = 'SHIPPING_INFO';

export const getProducts = (payload) => ({
        type: GET_PRODUCTS,
        payload: payload
});

export const shippingInfo = (payload) => ({
        type: SHIPPING_INFO,
        payload: payload
});

const initialState = {
    product: [],
    userAddress: []
};

export const productsReducer = (state = initialState, action) => {
    const {type, payload} = action;
    switch (type) {
        case SET_PRODUCTS:
            return{...state, product: payload};

        case SHIPPING_INFO:
            return {...state, userAddress: [...state.userAddress, payload]};

        default: return state
    }
};

//Saga
function* getProductsSaga() {
    while (true){
        try {
            yield take(GET_PRODUCTS);
            const req = yield fetch('https://demo3830727.mockable.io/products');
            const res = yield req.json();
            yield put({
                type: SET_PRODUCTS,
                payload: res
            })
        } catch (e) {
            console.log(e);
        }
    }
}

function* shippingInfoSaga() {
    while (true){
        try {
            const { payload } = yield take(SHIPPING_INFO);
            const response = yield fetch('https://demo3830727.mockable.io/info', {
                    method: 'POST',
                    headers: {
                        "Content-Type": "application/json"
                    },
                    body: JSON.stringify(payload)
                }
        );
            console.log(response.status);
        } catch (e) {
            console.log(e);
        }
    }
}

export function* rootSaga() {
    yield all([
        getProductsSaga(),
        shippingInfoSaga()
    ])
}

