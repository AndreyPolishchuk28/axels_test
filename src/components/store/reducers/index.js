import {createStore, combineReducers, applyMiddleware} from "redux";
import createSagaMiddleware from "redux-saga";
import {getProductsSaga} from "../action";
import {SET_PRODUCTS} from "../action";
import {all} from 'redux-saga/effects'

const initialState = {
    products: []
};

const productsReducer = (state = initialState, action) => {
    const {type, payload} = action;
    switch (type) {
        case SET_PRODUCTS:{
            return{
                ...state,
                products: payload
            }
        }
        default: return state
    }
};

function* rootSaga() {
    yield all([
        getProductsSaga()
    ])
}

const sagaMiddleware = createSagaMiddleware();
const reducer = combineReducers({
    product: productsReducer
});

export const store = createStore(reducer, applyMiddleware(sagaMiddleware));
sagaMiddleware.run(rootSaga);





