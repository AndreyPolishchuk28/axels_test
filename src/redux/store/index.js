import {createStore, applyMiddleware, combineReducers} from 'redux';
import createSagaMiddleware from 'redux-saga';

import { rootSaga } from '../ducks/products';
import { productsReducer } from '../ducks/products';

const sagaMiddleware = createSagaMiddleware();

const saveToLocalStorage = state => {
    try {
        const serializedState = JSON.stringify(state);
        localStorage.setItem('state', serializedState);
    } catch (e) {
        console.log(e);
    }
};

const loadFromLocalStorage = () => {
    try {
        const serializedState = localStorage.getItem('state');
        if(serializedState === null) return undefined;
        return JSON.parse(serializedState);
    } catch (e) {
        console.log(e);
        return undefined;
    }
};
const persistedState = loadFromLocalStorage();

export const store = createStore(combineReducers({
    products: productsReducer
}), persistedState, applyMiddleware(sagaMiddleware));

store.subscribe(() => saveToLocalStorage(store.getState()));
sagaMiddleware.run(rootSaga);
