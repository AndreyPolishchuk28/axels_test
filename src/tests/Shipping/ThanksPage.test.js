import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';

import { ThanksPage } from '../../components';
import { store } from '../../redux/store';

describe('ThanksPage', () => {
    it('should render ThanksPage component', () => {
        const component = renderer.create(
            <Provider store={store}>
                <Router>
                    <ThanksPage />
                </Router>
            </Provider>
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
