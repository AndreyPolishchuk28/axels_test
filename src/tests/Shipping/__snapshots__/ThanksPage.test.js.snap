// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThanksPage should render ThanksPage component 1`] = `
<div
  className="sc-AxjAm emnRQd container-fluid"
>
  <div
    className="d-flex justify-content-center bg row"
  >
    <div
      className="bg-white wrapper-shipping-width col-md-6 col-sm-8 col-10"
    >
      <div
        className="row"
      >
        <div
          className="text-left mt-5 ml-3 col"
        >
          <p
            className="sc-fzqBZW eOYARb"
          >
            Thank you for your order!
          </p>
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="text-left ml-3 col"
        >
          <p
            className="sc-fzqNJr bCRZog"
          >
            Order number is: 188787788
          </p>
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="text-left ml-3 mr-5 col"
        />
      </div>
      <div
        className="row"
      >
        <div
          className="text-left ml-3 col"
        >
          <p
            className="sc-fzoyAV fLPRJE mb-0"
          >
            Estimated delivery Day is
          </p>
          <p
            className="sc-fzoXzr hhhZcE"
          >
            Friday 1st April 2016
          </p>
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="text-left ml-3 mt-2 col"
        >
          <p
            className="sc-fzpjYC kJNrby"
          >
            Print Recipe
          </p>
        </div>
      </div>
    </div>
    <div
      className="grey wrapper-order-width col-md-5 col-sm-8 col-10"
    >
      <div
        className="sc-AxhCb gCHefG"
      >
        <div
          className="row"
        >
          <div
            className="col"
          >
            <h3
              className="sc-AxhUy ddBKQc"
            >
              Order Summary
            </h3>
          </div>
          <div
            className="text-right col"
          >
            <span
              className="sc-AxgMl OIChS"
            >
              edit order
            </span>
          </div>
        </div>
        <div
          className="wrapper-product ml-1 row"
        >
          <div
            className="d-flex flex-column justify-content-between text-left p-0 col-xl-8 col-md-8 col-sm-8 col-8"
          >
            <span
              className="sc-fzpans erUppB"
            >
              Subtotal
            </span>
            <span
              className="sc-fzpans erUppB"
            >
              Shipping
            </span>
            <span
              className="sc-fzpans erUppB"
            >
              Taxes
            </span>
          </div>
          <div
            className="text-right p-0 d-flex flex-column justify-content-between col-xl-4 col-md-4 col-sm-4 col-4"
          >
            <span
              className="sc-fzpans erUppB"
            >
              Free
            </span>
          </div>
        </div>
        <div
          className="mt-3 row"
        >
          <div
            className="col"
          >
            <span
              className="sc-fzplWN gbCGPk"
            >
              Total
            </span>
          </div>
          <div
            className="text-right pb-5 col"
          />
        </div>
        <div
          className="text-center mt-5 row"
        >
          <div
            className="col"
          >
            <p
              className="sc-fznyAO gGyRlb"
            >
              All purchases are subject to our 
              <span
                className="sc-fznKkj cYYIqp"
              >
                Terms and Conditions
              </span>
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
